### ISSUE TITLE
"unexpected EOF" error with the RPC client

### ISSUE THREAD

**Issue by RescueRaiders** [0xProject/0x-mesh#870]
Hello. I am running Mesh inside of Docker on an iMac (Catalina 10.15.6). The server seems to be working well. 
Here is the command line:
docker run --restart unless-stopped -p 60556:60556 -p 60557:60557 -p 60558:60558 -p 60559:60559  -e WS_RPC_ADDR=localhost:60557  -e ETHEREUM_CHAIN_ID="1" -e ETHEREUM_RPC_URL="MyEthereumRPC" -e VERBOSITY=5 -v ~/0xMeshTest/0x_mesh:/usr/mesh/0x_mesh 0xorg/mesh:9.4.0
I am running a simple golang RPC client on the same iMac, in another terminal. I tried get-stats or subscribe-to-orders, but it's always failing with an "EOF" error
go run ./examples/go/get-stats/main.go 
{"error":"unexpected EOF","level":"fatal","msg":"could not create client","time":"2020-07-20T09:48:17+02:00"}
exit status 1
(go version is 1.14.5 darwin/amd64)
I am seeing the same results with the Javascript example package.
Note that the below command works:
nc -zv 127.0.0.1 60557 returns Connection to 127.0.0.1 port 60557 [tcp/*]  succeeded
which means the server seems reachable and functional. 
It seems it's the rpc.Dial(addr) from the Ethereum RPC package returning this "unexpected EOF" error
Let me know if there is any other logs I could look at, or higher log verbosity. Perhaps a local network capture ?
Any idea ?
Thanks

**Comment by albrow** [comment]
> I am running a simple golang RPC client on the same iMac, in another terminal. I tried get-stats or subscribe-to-orders, but it's always failing with an "EOF" error

Can you share the code that you are running on the client side?

**Comment by RescueRaiders** [comment]
Sure. Actually, it's almost a copy/paste from 0x-mesh github: examples/go/subscribe-to-orders
which also fails with the same error at the client creation step.
Thanks.

// +build !js
  
// demo/get-stats is a short program that adds an order to 0x Mesh via RPC
package main

import (
        "fmt"
        "encoding/json"
        "github.com/0xProject/0x-mesh/rpc"
        "github.com/plaid/go-envvar/envvar"
        log "github.com/sirupsen/logrus"
)

type clientEnvVars struct {
        // RPCAddress is the address of the 0x Mesh node to communicate with.
        WSRPCAddress string `envvar:"WS_RPC_ADDR"`
}

func main() {
        log.SetFormatter(&log.JSONFormatter{})

        env := clientEnvVars{}
        if err := envvar.Parse(&env); err != nil {
                panic(err)
        }

        client, err := rpc.NewClient(env.WSRPCAddress)
        if err != nil {
                log.WithError(err).Fatal("could not create client")
        }

        stats, err := client.GetStats()
        if err != nil {
                log.WithError(err).Fatal("Couldn't get Stats")
        }
        statsJSON, err := json.MarshalIndent(stats, "<prefix>", "<indent>")
        if err != nil {
                log.Fatalf(err.Error())
        }
        fmt.Println(string(statsJSON))
}

**Comment by RescueRaiders** [comment]
I solved my issue. I had to run the docker instance with "-e WS_RPC_ADDR=:60557" and NOT "-e WS_RPC_ADDR=localhost:60557".
I found the explanation here: https://stackoverflow.com/questions/54101508/how-do-you-dockerize-a-websocket-server
I am fairly new to docker, so I am not fully understanding this.
Is it specific to the Mac ? (which uses a VM for running docker) 
In any case, your guide here https://0x-org.gitbook.io/mesh/getting-started/deployment is relying on the default value for WS_RPC_ADDR which if it contains localhost will likely leads to failure for a mac user to connect to the RPC socket.
I think this can be closed but interested to know your thoughts on this finding if you have time.
In any case, it now works for me on a Mac and I no longer get the "EOF" error which was likely due to a timeout on the client side.
